From 889b832224e712716510b28c83b72f70f7eb8f24 Mon Sep 17 00:00:00 2001
From: Jesse Schwartzentruber <truber@mozilla.com>
Date: Thu, 11 Mar 2021 11:08:36 -0500
Subject: [PATCH] glcpp: Fix undefined behaviour in glcpp

Specifically, fix this error (which is covered in existing tests):

    ../src/compiler/glsl/glcpp/pp.c:198:28: runtime error: applying non-zero offset 1 to null pointer
    SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/compiler/glsl/glcpp/pp.c:198:28 in
---
 src/compiler/glsl/glcpp/pp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/compiler/glsl/glcpp/pp.c b/src/compiler/glsl/glcpp/pp.c
index aadf899a7d0..db53e7dd165 100644
--- a/src/compiler/glsl/glcpp/pp.c
+++ b/src/compiler/glsl/glcpp/pp.c
@@ -195,11 +195,11 @@ remove_line_continuations(glcpp_parser_t *ctx, const char *shader)
 			}
 		}
 
-		search_start = backslash + 1;
-
 		if (backslash == NULL)
 			break;
 
+		search_start = backslash + 1;
+
 		/* At each line continuation, (backslash followed by a
 		 * newline), copy all preceding text to the output, then
 		 * advance the shader pointer to the character after the
-- 
GitLab

